kotlin/samples/recyclerview/src/main/java/com/squareup/sample/recyclerview/editablelistworkflow/RowValue.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.recyclerview.editablelistworkflow
+
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.squareup.sample.recyclerview.inputrows.InputRow
+
+/**
+ * Holds an [InputRow] and the current value to display in that row.
+ *
+ * Also has [writeValue] and [setListener], which are convenience wrappers around the methods of
+ * the same name on [InputRow] that get around type constraints and do the casting for you.
+ */
+data class RowValue<T, VH : ViewHolder>(
+  val row: InputRow<T, VH>,
+  val value: T
+) {
+  constructor(row: InputRow<T, VH>) : this(row, row.defaultValue)
+
+  @Suppress("UNCHECKED_CAST")
+  fun writeValue(
```

Comment:
will it make sense to have generic methods here instead of casting everything to desired type?

Reply:
These methods actually exist only to get around the generics. Because we don't have specific types where we're using them, the constraints resolve to `Nothing`, so we need methods like this to do the casting.